homeassistant/components/mqtt/alarm_control_panel.py

```diff
@@ -81,6 +91,9 @@
 
 PLATFORM_SCHEMA_MODERN = MQTT_BASE_SCHEMA.extend(
     {
+        vol.Required(
```

Comment:
I do not think we should make this option required, because of the default value this is not needed.

Reply:
I debated that myself, wondering about the differences between Optional w/ default vs. Required w/ default. From a user configuration perspective they seemed equivalent.

However, I opted into Required because technically it _is_ actually required for the code to work: we do `self._config[CONF_SUPPORTED_FEATURES]` later on, rather than `self._config.get(CONF_SUPPORTED_FEATURES)` etc. Static analysis like mypy isn't smart enough to pick this up, but it could in the future. (git grepping Required.*default results into 384 matches across the tree, so it's definitely not unheard of either)

Reply:
Introducing a new Required option is weird as it is absent in all configurations from the user perspective. Because it has a default value, it is not required in the user config or discovery payload. Technically you are right that it does not make a difference.

Reply:
Could you help me understand why it's weird? If you can elaborate on what you fear would happen, or any references, or prior discussions.

My instincts suggest this is better for the reasons I stated, but given I'm a new contributor, my instincts are not fine-tuned for this codebase :) But when I tried looking to get a better feel of how others have approached this, I found lots of callsites doing it this way, even in very popular or core parts of HA, such as for example the Integration helper (CONF_ROUND_DIGITS, CONF_METHOD, CONF_UNIT_PREFIX, CONF_UNIT_TIME), Bluetooth (CONF_PASSIVE), ZHA, ZWave-js, etc. Would love to hear more to make this a learning experience for me.

Reply:
It is not clear to the users to make it required while it is not. It should be consistent with current configs that it is not needed to add the option. I cannot make it more clear than that.

Reply:
OK, let me try again: my understanding is that `vol.Required` = "_this code requires this attribute to function_", rather than "_the user needs to specify this option_". In fact given we provide a default here, a user-provided value is explicitly NOT required here. Is my understanding incorrect?

As for consistency, I already gave several examples in the codebase of options that are not explicitly needed, that are configured in the same way. So I don't understand the consistency argument. Could you help explain how this option is different than e.g. the Integration configuration options I mentioned above?

Reply:
We only use `Required` when there is no default and the schema will fail without it, so hat is why it needs to change. The required flag in the documentation needs to be inline with the code. It is hard to explain why an option is marked as `Required` while it is not. That is an unneeded restriction. Hope it is clear enough.

Reply:
As I mentioned, there are multiple counter-examples in the tree, so the statements you made ("_We only use `Required` when there is no default_") do not seem to match the existing practice, at least not consistently. A mere `git grep 'vol\.Required.*default'` is enough to validate this.

All that said, I think we've both spent way more time on this than it deserved, so I'll do as you asked.

Should I also send a PR to https://developers.home-assistant.io/docs/development_validation/ under the "Some things to keep in mind" section that says something like "_Do not ever use `vol.Required` with a default setting; use vol.Optional instead_"? Happy for you to tweak the wording as you please, and even happier if you'd like to submit that PR yourself. Just looking out for everyone's else time on this :)